plugins/outputs/newrelic/newrelic_test.go

```diff
@@ -0,0 +1,100 @@
+package newrelic
+
+import (
+	"testing"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/newrelic/newrelic-telemetry-sdk-go/telemetry"
+	"github.com/stretchr/testify/require"
+)
+
+func TestBasic(t *testing.T) {
+	nr := &NewRelic{
+		MetricPrefix: "Test",
+		InsightsKey:  "12345",
+	}
+
+	err := nr.Connect()
+	require.NoError(t, err)
+
+	err = nr.Write(testutil.MockMetrics())
+	require.NoError(t, err)
+}
+
+func TestNewRelic_Write(t *testing.T) {
+	type fields struct {
+		harvestor    *telemetry.Harvester
+		InsightsKey  string
+		MetricPrefix string
+	}
+	type args struct {
+		metrics []telegraf.Metric
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		metrics []telegraf.Metric
+		wantErr bool
+	}{
+		// TODO: Add test cases.
+		{
+			name: "Test: Basic mock metric write",
+			fields: fields{
+				InsightsKey:  "insightskey",
+				MetricPrefix: "test1",
+			},
+			metrics: testutil.MockMetrics(),
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			nr := &NewRelic{
+				harvestor:    tt.fields.harvestor,
+				InsightsKey:  tt.fields.InsightsKey,
+				MetricPrefix: tt.fields.MetricPrefix,
+			}
+			if err := nr.Write(tt.metrics); (err != nil) != tt.wantErr {
+				t.Errorf("NewRelic.Write() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func TestNewRelic_Connect(t *testing.T) {
+
+	tests := []struct {
+		name     string
+		newrelic *NewRelic
+		wantErr  bool
+	}{
+		{
+			name: "Test: No Insights key",
+			newrelic: &NewRelic{
+				MetricPrefix: "prefix",
+			},
+			wantErr: true,
+		},
+		{
+			name: "Test: Insights key",
+			newrelic: &NewRelic{
+				InsightsKey:  "12312133",
+				MetricPrefix: "prefix",
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			nr := &NewRelic{
+				harvestor:    tt.newrelic.harvestor,
+				InsightsKey:  tt.newrelic.InsightsKey,
+				MetricPrefix: tt.newrelic.MetricPrefix,
+			}
+			if err := nr.Connect(); (err != nil) != tt.wantErr {
```

Comment:
If this actually connects to a remote service, we should make the test be skipped in `-short` mode.

Reply:
@danielnelson This test is more of a negative test and only validates that we fail if we provide dummy values. Should we still skip it?

Reply:
@hsinghkalsi Is there a way to make this test seem more deterministic.  If the connection will always fail when run as a unit test, it may make sense to omit the Connect() since it will always fail or comment somewhere that fail is expected in this scenario.

@danielnelson as a side note, we are currently working on a blog post to highlight a really nice use case of this new plugin.   If we can get this merged in the next couple of days, when would we possible see this plugin in a released version of Telegraf?

Reply:
@danielnelson @jimhagan skipped the test in short mode.